File: app/src/main/java/ca/judacribz/gainzassist/models/db/ExerciseDao.java
package ca.judacribz.gainzassist.models.db;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.OnConflictStrategy;
import androidx.room.Query;
import androidx.room.Update;

import java.util.List;

import ca.judacribz.gainzassist.models.Exercise;

@Dao
public interface ExerciseDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    Long insert(Exercise exercise);

    @Query("SELECT * from exercises WHERE workout_id = :workoutId")
    LiveData<List<Exercise>> getLiveFromWorkout(long workoutId);

    @Query("SELECT * from exercises WHERE workout_id = :workoutId ORDER BY exercise_number")
    List<Exercise> getFromWorkout(long workoutId);

    @Query("SELECT * from exercises WHERE id = :id")
    LiveData<Exercise> getLiveExerciseFromWorkout(long id);

    @Query("SELECT * from  exercises WHERE id = :id")
    LiveData<Exercise> getLive(long id);

    @Query("SELECT * from  exercises WHERE id = :id")
    Exercise get(long id);

    @Query("SELECT DISTINCT name FROM exercises")
    LiveData<List<String>> getAllUniqueNames();

    @Query("SELECT id from  exercises WHERE name = :name AND workout_id = :workoutId")
    long getId(String name, long workoutId);

    @Query("UPDATE exercises SET weight = :weight WHERE id = :id")
    void updateWeight(float weight, long id);

    @Update
    void update(Exercise... exercise);


    @Delete
    void delete(Exercise... exercises);
}

File: app/src/main/java/ca/gainzassist/models/db/ExerciseDao.kt
package ca.gainzassist.models.db

import androidx.lifecycle.LiveData
import androidx.room.Dao
import androidx.room.Delete
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import androidx.room.Update
import ca.gainzassist.models.Exercise

@Dao
interface ExerciseDao {

    @Insert(onConflict = OnConflictStrategy.REPLACE)
    fun insert(exercise: Exercise): Long

    @Query("SELECT * from exercises WHERE workout_id = :workoutId")
    fun getLiveFromWorkout(workoutId: Long): LiveData<List<Exercise>>

    @Query("SELECT * from exercises WHERE workout_id = :workoutId ORDER BY exercise_number")
    fun getFromWorkout(workoutId: Long): List<Exercise>

    @Query("SELECT * from exercises WHERE id = :id")
    fun getLiveExerciseFromWorkout(id: Long): LiveData<Exercise>

    @Query("SELECT * from  exercises WHERE id = :id")
    fun getLive(id: Long): LiveData<Exercise>

    @Query("SELECT * from  exercises WHERE id = :id")
    fun get(id: Long): Exercise?

    @Query("SELECT DISTINCT name FROM exercises")
    fun getAllUniqueNames(): LiveData<List<String>>

    @Query("SELECT id from  exercises WHERE name = :name AND workout_id = :workoutId")
    fun getId(name: String, workoutId: Long): Long

    @Query("UPDATE exercises SET weight = :weight WHERE id = :id")
    fun updateWeight(weight: Float, id: Long)

    @Update
    fun update(vararg exercise: Exercise)

    @Delete
    fun delete(vararg exercises: Exercise)
}
